OpenEdXMobile/src/main/java/org/edx/mobile/view/adapters/NewCourseOutlineAdapter.java

```diff
@@ -159,38 +159,32 @@ public final View getView(int position, View convertView, ViewGroup parent) {
         final int type = getItemViewType(position);
 
         // FIXME: Re-enable row recycling in favor of better DB communication [MA-1640]
-        //if (convertView == null) {
-        switch (type) {
-            case SectionRow.ITEM: {
-                convertView = inflater.inflate(R.layout.row_course_outline_list, parent, false);
-                // apply a tag to this list row
-                ViewHolder tag = getTag(convertView);
-                convertView.setTag(tag);
-                return getRowView(position, convertView);
-            }
-            case SectionRow.SECTION: {
-                convertView = inflater.inflate(R.layout.row_section_header, parent, false);
-                return getHeaderView(position, convertView);
-            }
-            case SectionRow.COURSE_CARD: {
-                if (convertView == null) {
+        // INITIALIZATION
+        if (convertView == null) {
+            switch (type) {
+                case SectionRow.ITEM: {
+                    convertView = inflater.inflate(R.layout.row_course_outline_list, parent, false);
+                    // apply a tag to this list row
+                    ViewHolder tag = getTag(convertView);
```

Comment:
We should apply view holder pattern to all types of SectionRow.

Reply:
Let's defer these improvements for now to the tech debt ticket and focus on issues at hand.